tests/ORM/ManyToManySingleEntityTest.php

```diff
@@ -157,4 +157,39 @@ public function testClearAndFillRelation(): void
 
         self::assertTrue(true);
     }
+
+    public function testNotTriggersRehydrate(): void
+    {
+        $role = new RbacRole('superAdmin', 'description');
+
+        $permission = new RbacPermission('writeUser');
+
+        $role->children->add($permission);
+        $permission->parents->add($role);
+
+        $tr = new Transaction($this->orm);
+        $tr->persist($role);
+        $tr->run();
+
+        unset($role, $permission);
+
+        $this->orm = $this->orm->withHeap(new Heap());
+
+        /** @var RbacRole $fetchedRole */
+        $fetchedRole = (new Select($this->orm, 'rbac_item'))->wherePK('superAdmin')->fetchOne();
+        /** @var RbacPermission $fetchedPermission */
+        $fetchedPermission = (new Select($this->orm, 'rbac_item'))->wherePK('writeUser')->fetchOne();
+
+        $fetchedRole->description = 'updated description';
+
+        // unlink
+        $fetchedRole->children->removeElement($fetchedPermission);
+        $fetchedPermission->parents->removeElement($fetchedRole);
+
+        self::assertSame('updated description', $fetchedRole->description);
+
+        $tr = new Transaction($this->orm);
+        $tr->persist($fetchedRole);
+        $tr->run();
```

Comment:
```suggestion
        $this->save($fetchedRole);
```

If here you check that no exceptions will be thrown, then write about it in the comments.
You can also check the number of read or write requests by methods:
```php
        $this->captureWriteQueries();
        $this->captureReadQueries();

        $this->save($fetchedRole);

        $this->assertNumWrites(2);
        $this->assertNumReads(0);
```

Reply:
No, I'm checking that role description will not be overwritten with current database value after loading permission parents.

Second save called for `BaseTest::assertEntitySynced()` not throw exceptions.

Reply:
> Second save called for BaseTest::assertEntitySynced() not throw exceptions.

Then this instead:

```php
$this->orm = $this->orm->withHeap(new Heap());
```

Reply:
Thx. Done